engine/src/main/java/org/entando/entando/aps/system/services/digitalexchange/client/DigitalExchangeCall.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2018-Present Entando Inc. (http://www.entando.com) All rights reserved.
+ * 
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ * 
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+package org.entando.entando.aps.system.services.digitalexchange.client;
+
+import java.util.List;
+import org.entando.entando.aps.system.services.digitalexchange.model.DigitalExchange;
+import org.entando.entando.web.common.model.RestResponse;
+import org.springframework.core.ParameterizedTypeReference;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.web.util.UriComponentsBuilder;
+
+/**
+ * Contains all the information necessary for querying a specific Digital
+ * Exchange endpoint (HTTP method, response type, path) and provides the logic
+ * necessary for combining multiple responses provided by the same endpoint of
+ * different Digital Exchange instances.
+ *
+ * @param <R> the type of each DE response
+ * @param <C> the type of the combined response
+ */
+public abstract class DigitalExchangeCall<R extends RestResponse, C> {
+
+    private final HttpMethod method;
+    private final ParameterizedTypeReference parameterizedTypeReference;
```

Comment:
If I understood the class correctly, this would be:
`    private final ParameterizedTypeReference<R> parameterizedTypeReference;
`